src/components/MainTable/index.js

```diff
@@ -0,0 +1,188 @@
+import { React, useState, useEffect} from 'react';
+import { useSelector, useDispatch } from 'react-redux'
+import { v4 as uuid } from 'uuid';
+import { getCountries, showDisplayedCountries, pin, unpin, deleteFromDisplay, deleteFromPinned, getPinned } from '../../store/countries/actions';
+import CustomButton from '../customButton/index';
+import CustomInput from '../customInput/index';
+import CountryCard from '../CountryCard';
+import './style.scss'
+import Flex from './../../assets/styledComponents/Flex';
+
+const MainTable = () => {
+    const dispatch = useDispatch()
+    const [input, setInput] = useState('')
+    const [currentCard, setCurrentCard] = useState({})
+
+    const {countries, displayedCountries, pinnedCountries}= useSelector(state => state.countryReducer)
+
+    useEffect(() => {
+        if ( !JSON.parse(sessionStorage.getItem('display')) ) {
+            sessionStorage.setItem('display', '[]')
+        } if (!JSON.parse(sessionStorage.getItem('pinned') )){
+            sessionStorage.setItem('pinned', '[]')
+        }
+
+        dispatch(getCountries())
+        dispatch(showDisplayedCountries(JSON.parse(sessionStorage.getItem('display'))))
+        dispatch(getPinned(JSON.parse(sessionStorage.getItem('pinned'))))
+    }, [])
+
+    const displayCountry = () => {
```

Comment:
The right way is not to trigger this function from use effect  and declare this variable using useMemo